sql/api/src/main/scala/org/apache/spark/sql/Observation.scala

```diff
@@ -78,7 +78,11 @@ class Observation(val name: String) {
   @throws[InterruptedException]
   def get: Map[String, Any] = {
     val row = getRow
-    row.getValuesMap(row.schema.map(_.name))
+    if (row.schema == null) {
```

Comment:
1. We still wait indefinitely via `Duration.Inf`, don't we?

https://github.com/apache/spark/blob/4baf94d83f510c5a095a4383e7f53e4bd488fedb/sql/api/src/main/scala/org/apache/spark/sql/Observation.scala#L132-L134

2. When `row != null` and `row.schema == null`?